jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHttpOutputInterceptor.java

```diff
@@ -229,7 +229,15 @@ protected void commit(ByteBuffer content, boolean complete, Callback callback)
             LOG.debug("{} compressing {}", this, _deflater);
             _state.set(GZState.COMPRESSING);
 
-            gzip(content, complete, callback);
+            if (content.remaining() == 0)
+            {
+                // We are committing, and we have no content to compress.
+                _interceptor.write(content, complete, callback);
+            }
```

Comment:
Let's be very explicit that we are just writing the headers:

```suggestion
            if (BufferUtil.isEmpty(content))
            {
                // We are committing, but have no content to compress, so flush empty buffer to write headers.
                _interceptor.write(BufferUtil.EMPTY_BUFFER, complete, callback);
            }
```

I've also used BufferUtil to protect against a null content... probably would not happen, but there is only one other place in this method that assumes a non null content, so may as well protect.  So also change the `content.remaining` on 208 to `BufferUtil.remaining(content)`